pkg/kn/traffic/compute_test.go

```diff
@@ -51,169 +53,188 @@ func newTestTrafficCommand() (*cobra.Command, *flags.Traffic) {
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "", 100, true)),
+			[]string{"--untag", "latest"},
+			[]string{"@latest"},
+			[]string{""},
+			[]int{100},
+		},
+		{
+			"replace revision pointing to 'latest' tag from 'echo-v1' to 'echo-v2' revision",
+			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("latest", "echo-v1", 50, false), newTarget("", "echo-v2", 50, false)),
+			[]string{"--untag", "latest", "--tag", "echo-v1=old,echo-v2=latest"},
+			[]string{"echo-v1", "echo-v2"},
+			[]string{"old", "latest"},
+			[]int{50, 50},
+		},
 	} {
-		testCmd, tFlags := newTestTrafficCommand()
-		testCmd.SetArgs(testCase.inputFlags)
-		testCmd.Execute()
-		err, targets := Compute(testCmd, testCase.existingTraffic, tFlags)
-		if err != nil {
-			t.Fatal(err)
-		}
-		for i, target := range targets {
-			if testCase.desiredRevisions[i] == "@latest" {
-				assert.Equal(t, *target.LatestRevision, true)
-			} else {
-				assert.Equal(t, target.RevisionName, testCase.desiredRevisions[i])
+		t.Run(testCase.name, func(t *testing.T) {
+			if lper, lrev, ltag := len(testCase.desiredPercents), len(testCase.desiredRevisions), len(testCase.desiredTags); lper != lrev || lper != ltag {
+				t.Fatalf("length of desird revisions, tags and percents is mismatched: got=(desiredPercents, desiredRevisions, desiredTags)=(%d, %d, %d)",
+					lper, lrev, ltag)
+			}
+
+			testCmd, tFlags := newTestTrafficCommand()
+			testCmd.SetArgs(testCase.inputFlags)
+			testCmd.Execute()
+			err, targets := Compute(testCmd, testCase.existingTraffic, tFlags)
+			if err != nil {
+				t.Fatal(err)
+			}
+			for i, target := range targets {
+				if testCase.desiredRevisions[i] == "@latest" {
+					assert.Equal(t, *target.LatestRevision, true)
+				} else {
+					assert.Equal(t, target.RevisionName, testCase.desiredRevisions[i])
+				}
+				assert.Equal(t, target.Tag, testCase.desiredTags[i])
+				assert.Equal(t, target.Percent, testCase.desiredPercents[i])
 			}
-			assert.Equal(t, target.Tag, testCase.desiredTags[i])
-			assert.Equal(t, target.Percent, testCase.desiredPercents[i])
-		}
+		})
 	}
 }
 
 func TestComputeErrMsg(t *testing.T) {
 	for _, testCase := range []trafficErrorTestCase{
-		// error for --traffic for invalid split operation
 		{
+			"invalid format for `--traffic` option",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("", "", 100, true)),
 			[]string{"--traffic", "@latest=100=latest"},
 			"expecting the value format in value1=value2, given @latest=100=latest",
 		},
-		// error for --tag for invalid split operation
 		{
+			"invalid format for `--tag` option",
 			append(newServiceTraffic([]v1alpha1.TrafficTarget{}), newTarget("", "", 100, true)),
 			[]string{"--tag", "@latest="},
 			"expecting the value format in value1=value2, given @latest=",
 		},
-		// error for --traffic for repeating @latest
 		{
+			"repeatedly spliting traffic to '@latest' revision",
```

Comment:
and '' for @latest here.

I'd say lets remove both of them and use them as plain strings like --tag and @latest